src/AccountDeleter/Job.cs

```
@@ -235,5 +241,67 @@ protected override void ConfigureAutofacServices(ContainerBuilder containerBuild
                 .As<IEntityRepository<PackageOwnerRequest>>()
                 .InstancePerLifetimeScope();
         }
+
+        private void RegisterAuditingServices(IServiceCollection services)
+        {
+            if (IsDebugMode)
+            {
+                services.AddSingleton<AuditingService>(sp =>
+                {
+                    return GetAuditingServiceForLocalFileSystem();
+                });
+            }
+
+            services.AddSingleton<IAuditingService>(sp =>
+            {
+                var addInAuditingServices = GetAddInServices<IAuditingService>();
+                var auditingServices = new List<IAuditingService>(addInAuditingServices);
+
+                try
+                {
+                    auditingServices.Add(sp.GetRequiredService<AuditingService>());
+                }
+                catch
```

Swallowing all exceptions is generally discouraged.

What are the exceptions you're trying to handle?  Can you handle it more targetedly?  For example, when no default auditing service was registered, does `sp.GetRequiredService<AuditingService>()` throw or return `null`?  If the former, catch that specific exception.  If the latter, remove the blind `try...catch` and add a `null` check.

---

Updated this to catch InvalidOperationException as specified that this method throws by https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serviceproviderserviceextensions.getrequiredservice?view=aspnetcore-1.1